### PR TITLE
[PXS-66810] Put the ResultGMatchers code in its own module which refe…

### PR DESCRIPTION
**Pull request by acloudmovingby**
…rences the parent pom as its parent, but the directory is located inside of scala because Matt E said to do that.

### CODE REVIEW

File: scala/scalatest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" ?>
+<!--
+    Copyright 2012-2020 Paytronix Systems, Inc.
```
Comment (bebingando) [inline scala/scalatest/pom.xml:3]: new source file from scratch, should probably just use the current year in this copyright

**Review by bebingando — COMMENTED** [review]
you need to add a `<module>scala/scalatest</module>` to the modules list in the parent pom.xml or else your new code won't get built and packaged with the project

**Review by bebingando — CHANGES_REQUESTED** [review]
see my prior comment

File: scala/scalatest/src/main/scala/com/paytronix/utils/scalatest/ResultGMatchers.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+  def beFailedWithMessage[E, A](msg: String) =
+    Matcher { (left: ResultG[E, A]) =>
+      MatchResult(
+        left match {
+          case Okay(_) => false
+          case f@FailedG(t, e) => f.message == msg
+        },
+        s"$left did not fail with message \' $msg \'",
```
Comment (bebingando) [inline scala/scalatest/src/main/scala/com/paytronix/utils/scalatest/ResultGMatchers.scala:26]: why do you need to escape the single quotes?

Comment (acloudmovingby) [reply scala/scalatest/src/main/scala/com/paytronix/utils/scalatest/ResultGMatchers.scala:26]: No idea, I just copied and pasted what was there previously. I can change it if you'd like.

Comment (acloudmovingby) [reply scala/scalatest/src/main/scala/com/paytronix/utils/scalatest/ResultGMatchers.scala]: Sorry, that's a lame response and obviously anything I put in a PR is my responsibility. I tried it out on the REPL and you're right, they don't need to be escaped. I just removed the '\' characters and pushed the change.

**Review by bebingando — APPROVED** [review]

**Review by jthorntonpx — APPROVED** [review]

**Pull request merged**